modules/dockermodelrunner/internal/sdk/client/pull.go

```diff
@@ -31,6 +36,35 @@ func (c *Client) PullModel(ctx context.Context, fullyQualifiedModelName string)
 		return fmt.Errorf("unexpected status code: %d", resp.StatusCode)
 	}
 
+	namespace := strings.Split(fullyQualifiedModelName, "/")[0]
```

Comment:
suggestion: I think we just need to read the body to completion, no need for the retry which the caller can handle if desired.

So an io.Copy(io.Discard, resp.Body) should be all that's needed

Reply:
The cool thing of this is the logs providing progress

Reply:
you could do that by outputting each line received if needed.